tests/test_snapshot/test_metadata.py

```diff
@@ -46,11 +72,49 @@ def test_metadata_validate(metadata_kwargs: Dict, expected_error: Type[Exception
         "metadata_kwargs, metadata_dict, matches",
         [
             (DEFAULT_METADATA_KWARGS, {"arguments": FAKE_CALLER_INFO.args}, True),
-            (DEFAULT_METADATA_KWARGS, {"arguments": {"foo": 1, "bar": 2}}, False),
+            (
+                DEFAULT_METADATA_KWARGS,
+                {"arguments": {"foo": 1, "bar": 2, "foobar": [1, 2], "barfoo": [1, 2]}},
```

Comment:
suggest to add more type variation in the tests. perhaps make barfoo another nested dict or a set. Also could type check with 2 vs "2"

Reply:
these are getting checked against

`args={"foo": 1, "bar": "two", "foobar": [1, 2], "barfoo": array([1, 2])},`

so it is already doing a 2 vs "2" check. and foobar is a list while barfoo is an array. also test_pytest.py handles a list of dictionaries. still want me to add nested dict and set here?

Reply:
added a bunch more tests anyway 

[73eeb61](https://github.com/MORSECorp/snappiershot/pull/96/commits/73eeb6115264ded1ca45915f7bc655768ed238f1)